firebase-firestore/src/androidTest/java/com/google/firebase/firestore/testutil/IntegrationTestUtil.java

```diff
@@ -93,11 +111,33 @@ public static FirebaseFirestoreSettings newTestSettings() {
 
   public static FirebaseFirestoreSettings newTestSettingsWithSnapshotTimestampsEnabled(
       boolean enabled) {
-    return new FirebaseFirestoreSettings.Builder()
-        .setHost(provider.firestoreHost())
-        .setPersistenceEnabled(true)
-        .setTimestampsInSnapshotsEnabled(enabled)
-        .build();
+    FirebaseFirestoreSettings.Builder settings = new FirebaseFirestoreSettings.Builder();
+
+    if (CONNECT_TO_EMULATOR) {
+      settings.setHost(String.format("%s:%d", EMULATOR_HOST, EMULATOR_PORT));
+
+      // Disable SSL and hostname verification
```

Comment:
Fascinating!

This seems strange though. Don't we have a setting to disable SSL, specifically for this sort of test environment?  It may not actually work, but it seems like we should either make it work (by doing something like this?) or else remove that setting and make this the "correct" way to test against a non-ssl environment.

cc/ @allisonbm92 again.

Reply:
Done